File: src/main/java/info/szadkowski/contact/service/mail/MailMessageService.java
package info.szadkowski.contact.service.mail;

import info.szadkowski.contact.model.MessageRequest;
import info.szadkowski.contact.properties.MailAddressesProperties;
import info.szadkowski.contact.service.MessageService;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.MimeMessageHelper;

import javax.mail.internet.MimeMessage;

public class MailMessageService implements MessageService {
  private final MailAddressesProperties mailAddressesProperties;
  private final JavaMailSender javaMailSender;

  public MailMessageService(MailAddressesProperties mailAddressesProperties,
                            JavaMailSender javaMailSender) {
    this.mailAddressesProperties = mailAddressesProperties;
    this.javaMailSender = javaMailSender;
  }

  @Override
  public void send(MessageRequest message) {
    MimeMessage mimeMessage = javaMailSender.createMimeMessage();

    MimeMessageHelper helper = new MimeMessageHelper(mimeMessage);
    try {
      helper.setSubject(message.getSubject());
      helper.setFrom(mailAddressesProperties.getSenderMail());
      helper.setTo(mailAddressesProperties.getRecipientMail());
      helper.setText(message.getContent());

      javaMailSender.send(mimeMessage);
    } catch (Exception e) {
      throw new MessageSendException(e);
    }
  }
}

File: src/main/kotlin/info/szadkowski/contact/service/mail/MailMessageService.kt
package info.szadkowski.contact.service.mail

import info.szadkowski.contact.model.MessageRequest
import info.szadkowski.contact.properties.MailAddressesProperties
import info.szadkowski.contact.service.MessageService
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import org.springframework.mail.javamail.JavaMailSender
import org.springframework.mail.javamail.MimeMessageHelper

class MailMessageService(
    private val mailAddressesProperties: MailAddressesProperties,
    private val javaMailSender: JavaMailSender
) : MessageService {
    override suspend fun send(message: MessageRequest) {
        val helper = MimeMessageHelper(javaMailSender.createMimeMessage())

        try {
            withContext(Dispatchers.IO) {
                javaMailSender.send(helper.run {
                    setSubject(message.subject)
                    setFrom(mailAddressesProperties.senderMail)
                    setTo(mailAddressesProperties.recipientMail)
                    setText(message.content)

                    mimeMessage
                })
            }
        } catch (e: Exception) {
            throw MessageService.MessageSendException(e)
        }
    }
}
